pythonPackages.future: 0.15.2 -> 0.16.0

erictapen opened the pull request:
###### Motivation for this change
Needed a newer version for another package.

###### Things done
Bumped version and used fetchFromGitHub insted of fetchurl.

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

erictapen committed:
pythonPackages.future: 0.15.2 -> 0.16.0

Also moved the expression from python-packages.nix to ./pkgs/development/python-modules/future/default.nix due to discussion in #26220
Used fetchPypi insted of fetchurl.

FRidh commented:
Can you move this expression to `pkgs/development/python-modules/future/default.nix`.

>  and used fetchFromGitHub insted of fetchurl.

Why?

erictapen commented:
> Can you move this expression to pkgs/development/python-modules/future/default.nix.

Will edit this asap. Missed that policy.

> Why?

I felt like `fetchFromGitHub` would fit better, as it is more specific. Please correct me if you think that it is not good practice.

FRidh commented:
@erictapen with fetchPypi we can update the version and hash automatically (see maintainers/scripts/update-python-libraries)

erictapen commented:
Moved the expression to `./pkgs/development/python-modules/future/default.nix` and used `fetchPypi`.

Please note, that this is my first contribution besides a simple version bump. I do not really understand what I've done, I just changed things until my package compiled (and worked).

FRidh commented on pkgs/development/python-modules/future/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib
+, pkgs
```
don't pass in the whole package set, instead, use individual packages

FRidh commented on pkgs/development/python-modules/future/default.nix:
```diff
@@ -0,0 +1,39 @@
+    sha256 = "1nzy1k4m9966sikp0qka7lirh8sqrsyainyf8rk97db7nwdfv773";
+  };
+
+  propagatedBuildInputs = lib.optionals isPy26 (with pkgs.pythonPackages; [ importlib argparse ]);
```
We are not going to use the package set, but we're going to pass in the individual requirements. Therefore, in the top, you can add the parameters importlib and argparse and remove here `with pkgs.pythonPackages; `

FRidh reviewed (changes requested):
> Please note, that this is my first contribution besides a simple version bump. I do not really understand what I've done, I just changed things until my package compiled (and worked).

You've made a good start.   Just these two edits and then it should be ok to go in. Before, we called the function `buildPythonPackage` with a set of arguments directly in `python-packages.nix`. Now, we moved that expression to a separate file. That file is a function, returning our derivation. So, we call a function with a set of arguments and that function calls `buildPythonPackage` with a set of arguments.

erictapen commented:
Thanks alot. Did you mean this?

FRidh commented:
Yep!

cc maintainer @prikhi
